resolver/map.go

```diff
@@ -41,12 +41,8 @@ func NewAddressMap() *AddressMap {
 // find returns the index of addr in the addressMapEntry slice, or -1 if not
 // present.
 func (l addressMapEntryList) find(addr Address) int {
-	if len(l) == 0 {
-		return -1
-	}
 	for i, entry := range l {
-		if entry.addr.ServerName == addr.ServerName &&
-			entry.addr.Attributes.Equal(addr.Attributes) {
+		if entry.addr.equalWithAttributes(addr) {
```

Comment:
It's a bit unfortunate that we're checking `Addr` now, given that we know they are equal due to calling this method with this input.

Should we use a map key that has both `Addr` _and_ `ServerName` in order to make the map even more effective?  It's just the attributes that cause the problem of hashability.  (We could use an `Address` as the map key but only set the two fields we care about.)

Reply:
Changed the key to `resolver.Address` and got rid of `equalWithAttributes` and changed the code to only check equality of attributes here.